### PR TITLE
Add retry logic to groovy based api queries

### PR DESCRIPTION
**Pull request by M-Davies**
Issue: https://github.com/AdoptOpenJDK/openjdk-build/issues/1873
Signed-off-by: Morgan Davies <morgan.davies@ibm.com>

### CODE REVIEW

**Comment by M-Davies** [comment]
Draft for now as I don't want to disrupt the nightlies if this doesn't work. @karianna @johnoliver Can I get reviews in the meantime?

**Comment by johnoliver** [comment]
looks good to me

File: src/JobHelper.groovy [hunk 1]
```diff
@@ -79,6 +79,22 @@ class JobHelper {
...
-    public static getAvailableReleases() { return queryApi("https://api.adoptopenjdk.net/v3/info/available_releases") }
+    public static getAvailableReleases() {
+        def responseObject = null
+        for (int count = 0; count < 5; count++) {
+            try {
+                responseObject = queryApi("https://api.adoptopenjdk.net/v3/info/available_release")
+                break
+            } catch (RuntimeException e) {
+                println "Failed to query or parse the adopt api\nError:${e}\nRetrying in 30 seconds..."
+                sleep 30
```
Comment (johnoliver) [inline src/JobHelper.groovy:90]: does sleep not need to be imported?

Comment (johnoliver) [reply src/JobHelper.groovy:90]: and also make sure that is 30 seconds not milliseconds

Comment (M-Davies) [reply src/JobHelper.groovy:90]: It shouldn't be. Sleep is used [elsewhere](https://github.com/AdoptOpenJDK/openjdk-build/blob/master/pipelines/build/common/config_regeneration.groovy#L293) without an import

Comment (M-Davies) [reply src/JobHelper.groovy]: > and also make sure that is 30 seconds not milliseconds

Done

**Comment by M-Davies** [comment]
Kicked off https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk16-pipeline/10/console to test

**Pull request merged**